### PR TITLE
Use venv instead of virtualenv (fixes #188)

### LINKED ISSUE #188: Migrate from `virtualenv` to `-m venv`
The latter seems to be the prescribed method in Python3. Unfortunately, it does not create `sys.real_prefix`. Fortunately though, both setup `os.environ['VIRTUAL_ENV']`.

- [ ] Fix README to refer to `-m venv` and not mention `virtualenv` anywhere
- [ ] Change `common.mk` to use `os.environ['VIRTUAL_ENV']` for detecting a venv
- [ ] Test steps in README
- [ ] Update circleci/config.yml

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/AZUL-139)
┆Project Name: azul
┆Issue Number: AZUL-139

### PR DESCRIPTION
**Pull request by achave11**
Resolves #188

* Updated `virtualenv` testing mechanisms.
* Updated `venv` activation instructions.

### CODE REVIEW

**Comment by DailyDreaming** [comment]
Looks familiar...

https://github.com/DataBiosphere/toil/blob/master/Makefile#L275

**Comment by achave11** [comment]
@DailyDreaming thanks for the suggestion. That looks much cleaner!

File: scripts/envhook.py [hunk 1]
```diff
@@ -12,11 +12,10 @@ def main(argv):
...
     parser.add_argument('action', choices=['install', 'remove'])
     options = parser.parse_args(argv)
-    import site
-    if hasattr(sys, 'real_prefix'):
-        # virtualenv's site does not have getsitepackages()
-        link_dir = os.path.abspath(os.path.join(os.path.dirname(site.__file__), 'site-packages'))
-    else:
+    try:
+        # Read VIRTUAL_ENV variable true for running with venv.
+        link_dir = os.path.abspath(os.path.join(os.environ["VIRTUAL_ENV"], "lib", "python3.6", 'site-packages'))
```
[the remark this answers is not in the document]
Comment (hannes-ucsc) [reply scripts/envhook.py]: IOW. There are three cases: 1) virtualenv, 2) venv and 3) neither. Case 3 should raise the RuntimeError. Case 1 should be handled as it currently is. Case 2 should ideally be handled by calling getsitepackages(). If that doesn't work, case 3 could be handled like you coded it here.

File: scripts/envhook.py [hunk 2]
```diff
@@ -12,10 +12,15 @@ def main(argv):
...
     parser.add_argument('action', choices=['install', 'remove'])
     options = parser.parse_args(argv)
-    import site
-    if hasattr(sys, 'real_prefix'):
-        # virtualenv's site does not have getsitepackages()
-        link_dir = os.path.abspath(os.path.join(os.path.dirname(site.__file__), 'site-packages'))
+    if 'VIRTUAL_ENV' in os.environ:  # Confirm virtual environment is active.
+        import site
+        if hasattr(site, 'getsitepackages'):
+            # If `getsitepackages` is present in site are runningn with `venv` or without a virtual environment.
```
Comment (hannes-ucsc) [inline scripts/envhook.py]: Incomprehensible. Typo in `runningn`

Comment (hannes-ucsc) [reply scripts/envhook.py]: Consider back-quoting the word site.

Comment (achave11) [reply scripts/envhook.py]: Both plain Python and `venv` have `getsitepackages()`

**Comment by hannes-ucsc** [comment]
Note my change to the commit title and the missing line ending that I fixed.

**Pull request merged**